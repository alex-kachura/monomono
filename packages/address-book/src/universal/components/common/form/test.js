import React from 'react';
import { render } from '@testing-library/react';
import { renderProviders } from '@oneaccount/react-foundations/lib/test-utils';
import { DefaultThemeProvider } from '@beans/theme';
import Form from './';
import { Formik } from '@oneaccount/react-foundations';

const fields = [
  {
    id: 'address-id',
    name: 'address-id',
    type: 'text',
    label: 'address.fields.address-id.label',
    xtype: 'address',
    placeholder: 'address.fields.address-id.placeholder',
    valuePath: 'id',
    hidden: true,
    required: false,
  },
  {
    id: 'postcode',
    name: 'postcode',
    type: 'text',
    xtype: 'address',
    label: 'address.fields.postcode.label',
    placeholder: 'address.fields.postcode.placeholder',
    valuePath: 'postcode',
    required: true,
  },
  {
    id: 'address-line1',
    name: 'address-line1',
    type: 'text',
    xtype: 'address',
    label: 'address.fields.address-line1.label',
    placeholder: 'address.fields.address-line1.placeholder',
    valuePath: '1',
    required: true,
  },
  {
    id: 'address-line2',
    name: 'address-line2',
    type: 'text',
    xtype: 'address',
    label: 'address.fields.address-line2.label',
    placeholder: 'address.fields.address-line2.placeholder',
    valuePath: '2',
    required: false,
  },
  {
    id: 'address-line3',
    name: 'address-line3',
    type: 'text',
    xtype: 'address',
    label: 'address.fields.address-line3.label',
    placeholder: 'address.fields.address-line3.placeholder',
    valuePath: '3',
    required: false,
  },
  {
    id: 'town',
    name: 'town',
    type: 'text',
    xtype: 'address',
    label: 'address.fields.town.label',
    placeholder: 'address.fields.town.placeholder',
    valuePath: 'postTown',
    required: true,
  },
  {
    id: 'phone-number',
    name: 'phone',
    type: 'tel',
    label: 'pages.clubcard-address.fields.phone-number.label',
    placeholder: 'pages.clubcard-address.fields.phone-number.placeholder',
    required: true,
  },
];

function renderForm() {
  return render(
    renderProviders({
      appConfig: {
        getLocalePhrase: (key) => key,
      },
      children: (
        <DefaultThemeProvider>
          <Formik>
            <Form
              initialValues={{}}
              url={'/url'}
              onSubmit={jest.fn()}
              title={'pages.delivery-address.title'}
              fields={fields}
              csrf="super-secure-token"
              schema={{}}
            />
          </Formik>
        </DefaultThemeProvider>
      ),
    }),
  );
}

describe('[Component: Form]', () => {
  describe('[State: Default]', () => {
    it('should render correctly', () => {
      const component = renderForm();

      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
