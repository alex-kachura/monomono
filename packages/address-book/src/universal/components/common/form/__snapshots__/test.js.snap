// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: Form] [State: Default] should render correctly 1`] = `
<DocumentFragment>
  <h1
    class="base-components__BaseElement-sc-150tvch-0 sc-htoDjs KMYtd typography___styledElement-sc-186l0ce-0 hmbiuS"
  >
    pages.delivery-address.title
  </h1>
  <div
    class="sc-bZQynM gBCjJB"
  >
    <form
      autocomplete="off"
      method="POST"
      novalidate=""
    >
      <div
        class="address is-find-address"
      >
        <input
          aria-disabled="false"
          class="styled__Container-sc-1bf1p22-0 eZwXEO"
          hidden=""
          id="address-id"
          name="address-id"
          placeholder="address.fields.address-id.placeholder"
          type="text"
          value=""
        />
        <div
          class="address-field postcode sc-ifAKCX eKJWeb styled__Container-sc-1bf1p22-0 hTDNaU"
        >
          <label
            class="styled__StyledLabel-sc-28yvb4-0 bPkari"
            for="postcode"
          >
            address.fields.postcode.label 
          </label>
          <div
            class="base-components__BaseElement-sc-150tvch-0 beans-form-group__children styled__FormGroupChildren-wt4ift-2 bNvdOT"
          >
            <input
              aria-disabled="false"
              class="styled__Container-sc-1bf1p22-0 eZwXEO"
              id="postcode"
              name="postcode"
              placeholder="address.fields.postcode.placeholder"
              type="text"
              value=""
            />
            <button
              aria-disabled="false"
              class="styled__BaseButton-sc-3mnirm-0 find-address styled__PrimaryButton-sc-3mnirm-2 giYBMC"
              role="button"
              type="button"
            >
              <span
                class="base-components__BaseElement-sc-150tvch-0 styled__InnerContainer-sc-3mnirm-1 fUwmia"
              >
                <span
                  class="base-components__BaseElement-sc-150tvch-0 styled__BaseSpan-sc-3mnirm-4 beans-button__text styled__TextSpan-sc-3mnirm-5 eqKhmg"
                >
                  address.find-address
                </span>
              </span>
            </button>
          </div>
        </div>
        <button
          aria-disabled="false"
          class="styled__BaseButton-sc-3mnirm-0 manual-link styled__LinkButton-sc-3mnirm-13 gyNqJj"
          data-tracking="enter manual address"
          role="button"
          type="button"
        >
          <span
            class="base-components__BaseElement-sc-150tvch-0 styled__InnerContainer-sc-3mnirm-1 fUwmia"
          >
            <span
              class="base-components__BaseElement-sc-150tvch-0 styled__BaseSpan-sc-3mnirm-4 beans-button__text styled__TextSpan-sc-3mnirm-5 eqKhmg"
            >
              address.address-manually
            </span>
          </span>
        </button>
        <div
          class="address-field sc-bdVaJa jzpMvO styled__Container-sc-1bf1p22-0 hTDNaU"
        >
          <label
            class="styled__StyledLabel-sc-28yvb4-0 bPkari"
            for="address-line1"
          >
            address.fields.address-line1.label 
          </label>
          <div
            class="base-components__BaseElement-sc-150tvch-0 beans-form-group__children styled__FormGroupChildren-wt4ift-2 bNvdOT"
          >
            <input
              aria-disabled="false"
              class="styled__Container-sc-1bf1p22-0 eZwXEO"
              id="address-line1"
              name="address-line1"
              placeholder="address.fields.address-line1.placeholder"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="address-field sc-bdVaJa jzpMvO styled__Container-sc-1bf1p22-0 hTDNaU"
        >
          <label
            class="styled__StyledLabel-sc-28yvb4-0 bPkari"
            for="address-line2"
          >
            address.fields.address-line2.label 
            <span
              class="base-components__BaseElement-sc-150tvch-0 styled__OptionalSpan-wt4ift-1 fgdShb"
            >
              (optional)
            </span>
          </label>
          <div
            class="base-components__BaseElement-sc-150tvch-0 beans-form-group__children styled__FormGroupChildren-wt4ift-2 bNvdOT"
          >
            <input
              aria-disabled="false"
              class="styled__Container-sc-1bf1p22-0 eZwXEO"
              id="address-line2"
              name="address-line2"
              placeholder="address.fields.address-line2.placeholder"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="address-field sc-bdVaJa jzpMvO styled__Container-sc-1bf1p22-0 hTDNaU"
        >
          <label
            class="styled__StyledLabel-sc-28yvb4-0 bPkari"
            for="address-line3"
          >
            address.fields.address-line3.label 
            <span
              class="base-components__BaseElement-sc-150tvch-0 styled__OptionalSpan-wt4ift-1 fgdShb"
            >
              (optional)
            </span>
          </label>
          <div
            class="base-components__BaseElement-sc-150tvch-0 beans-form-group__children styled__FormGroupChildren-wt4ift-2 bNvdOT"
          >
            <input
              aria-disabled="false"
              class="styled__Container-sc-1bf1p22-0 eZwXEO"
              id="address-line3"
              name="address-line3"
              placeholder="address.fields.address-line3.placeholder"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="address-field sc-bdVaJa jzpMvO styled__Container-sc-1bf1p22-0 hTDNaU"
        >
          <label
            class="styled__StyledLabel-sc-28yvb4-0 bPkari"
            for="town"
          >
            address.fields.town.label 
          </label>
          <div
            class="base-components__BaseElement-sc-150tvch-0 beans-form-group__children styled__FormGroupChildren-wt4ift-2 bNvdOT"
          >
            <input
              aria-disabled="false"
              class="styled__Container-sc-1bf1p22-0 eZwXEO"
              id="town"
              name="town"
              placeholder="address.fields.town.placeholder"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa jzpMvO styled__Container-sc-1bf1p22-0 hTDNaU"
      >
        <label
          class="styled__StyledLabel-sc-28yvb4-0 bPkari"
          for="phone-number"
        >
          pages.clubcard-address.fields.phone-number.label 
        </label>
        <div
          class="base-components__BaseElement-sc-150tvch-0 beans-form-group__children styled__FormGroupChildren-wt4ift-2 bNvdOT"
        >
          <input
            aria-disabled="false"
            class="styled__Container-sc-1bf1p22-0 eZwXEO"
            id="phone-number"
            name="phone"
            placeholder="pages.clubcard-address.fields.phone-number.placeholder"
            required=""
            type="tel"
            value=""
          />
        </div>
      </div>
      <input
        name="_csrf"
        type="hidden"
        value=""
      />
      <button
        aria-disabled="false"
        class="styled__BaseButton-sc-3mnirm-0 sc-dnqmqq fIOvhp styled__PrimaryButton-sc-3mnirm-2 giYBMC"
        role="button"
        type="submit"
      >
        <span
          class="base-components__BaseElement-sc-150tvch-0 styled__InnerContainer-sc-3mnirm-1 fUwmia"
        >
          <span
            class="base-components__BaseElement-sc-150tvch-0 styled__BaseSpan-sc-3mnirm-4 beans-button__text styled__TextSpan-sc-3mnirm-5 eqKhmg"
          >
            Submit
          </span>
        </span>
      </button>
    </form>
  </div>
</DocumentFragment>
`;
